src/sqlfluff/core/rules/base.py

```diff
@@ -349,18 +349,37 @@ def _raw_slices_from_templated_slices(
 class RuleContext:
     """Class for holding the context passed to rule eval functions."""
 
-    segment: BaseSegment
```

Comment:
Reordered the fields so the ones that have a "natural" default value are at the bottom. Python "dataclasses" require them to be last. Providing defaults here allows us to reduce the long list of `crawl()` parameters and simplify the code (for human readability).